src/types/sequence.rs

```diff
@@ -289,15 +302,33 @@ where
     Ok(v)
 }
 
+fn get_sequence_abc(py: Python<'_>) -> &PyType {
+    SEQUENCE_ABC
+        .get_or_init(py, || {
+            py.import("collections.abc")
+                .expect("coud not import 'collections.abc'")
+                .getattr("Sequence")
+                .expect("coud not access 'Sequence' from 'collections.abc'")
+                .downcast::<PyType>()
+                .expect("could not access 'collections.abc.Sequence'")
+                .into_py(py)
+        })
+        .as_ref(py)
+}
+
 impl<'v> PyTryFrom<'v> for PySequence {
+    /// Downcasting to `PySequence` requires the concrete class to be a subclass (or registered
+    /// subclass) of `collections.abc.Sequence` (from the Python standard library) - i.e.
+    /// `isinstance(<class>, collections.abc.Sequence) == True`.
     fn try_from<V: Into<&'v PyAny>>(value: V) -> Result<&'v PySequence, PyDowncastError<'v>> {
         let value = value.into();
-        unsafe {
-            if ffi::PySequence_Check(value.as_ptr()) != 0 {
-                Ok(<PySequence as PyTryFrom>::try_from_unchecked(value))
-            } else {
-                Err(PyDowncastError::new(value, "Sequence"))
-            }
+        if value
+            .is_instance(get_sequence_abc(value.py()))
+            .unwrap_or(false)
```

Comment:
Is `unwrap_or(false)` correct here? I would have thought it would be better to just return the error.

Reply:
This would make sense but I wasn't sure how. Do I need to implement `From<err::PyErr>` for `PyDowncastError`?

Reply:
Oh I see, we can't express that with the current return type. Maybe just leave a TODO in the code and we can figure that out another time?

Reply:
_That_ I know how to implement haha.